packages/http-client-csharp/emitter/src/emitter.ts

```diff
@@ -150,25 +152,40 @@ export async function $onEmit(context: EmitContext<NetEmitterOptions>) {
         const command = `dotnet --roll-forward Major ${generatorPath} ${outputFolder} -p ${options["plugin-name"]}${constructCommandArg(newProjectOption)}${constructCommandArg(existingProjectOption)}${constructCommandArg(debugFlag)}`;
         Logger.getInstance().info(command);
 
-        const result = await execAsync(
-          "dotnet",
-          [
-            "--roll-forward",
-            "Major",
-            generatorPath,
-            outputFolder,
-            "-p",
-            options["plugin-name"],
-            newProjectOption,
-            existingProjectOption,
-            debugFlag,
-          ],
-          { stdio: "inherit" },
-        );
-        if (result.exitCode !== 0) {
-          if (result.stderr) Logger.getInstance().error(result.stderr);
-          if (result.stdout) Logger.getInstance().verbose(result.stdout);
-          throw new Error(`Failed to generate SDK. Exit code: ${result.exitCode}`);
+        try {
+          const result = await execAsync(
+            "dotnet",
+            [
+              "--roll-forward",
+              "Major",
+              generatorPath,
+              outputFolder,
+              "-p",
+              options["plugin-name"],
+              newProjectOption,
+              existingProjectOption,
+              debugFlag,
+            ],
+            { stdio: "inherit" },
+          );
+          if (result.exitCode !== 0) {
+            if (result.stderr) Logger.getInstance().error(result.stderr);
+            if (result.stdout) Logger.getInstance().verbose(result.stdout);
+            throw new Error(`Failed to generate SDK. Exit code: ${result.exitCode}`);
+          }
+        } catch (error: any) {
+          if (error && "code" in (error as {}) && error["code"] === "ENOENT") {
```

Comment:
Is this too general of a condition? Is there a way to make it specific to dotnet being missing?

Reply:
Also, don't we need to ensure that .NET 8 is installed? What would happen with the current approach if .NET 6 is installed.

Reply:
hello @JoshLove-msft as discussed before, we will not do runtime time check every time when we generate sdk, we just check the error if it is failed because of missing .NET and give out a nice error message. 
And we ask .NET 8 in the message to consistent with the requirement we write in the Readme.md of the library.

Reply:
Discussed in the meeting, we will do runtime check when generation fail, and identify if is because of dotnet missing. I will update it.